shotover-proxy/src/codec/cassandra.rs

```diff
@@ -89,6 +110,70 @@ impl Decoder for CassandraCodec {
     }
 }
 
+fn get_use_keyspace(message: &mut Message) -> Option<Identifier> {
+    if let Some(Frame::Cassandra(frame)) = message.frame() {
+        if let CassandraOperation::Query { query, .. } = &mut frame.operation {
+            if let CassandraStatement::Use(keyspace) = query.as_ref() {
+                return Some(keyspace.clone());
+            }
+        }
+    }
+    None
+}
+
+fn set_default_keyspace(message: &mut Message, keyspace: &Identifier) {
+    // TODO: rewrite Operation::Prepared in the same way
```

Comment:
Not sure what this means. What needs to be done for `Operation::Prepared`?

Reply:
The cassandra server will apply the current `use`d keyspace to Operation::Prepared messages that it receives.
To make that visible to transforms we should apply the same modifications that we apply to queries and batches to prepared statements as well

Reply:
Yes I remember reading something interesting about prepared queries and keyspaces but I can't find it now. IIRC:

- it applies the used keyspace to `PREPARE` statements
- The keyspace set when the `PREPARE` statement was sent takes precedence over the current used keyspace (if it has changed)

Reply:
yep, that matches my understanding too